contracts/validators/PhaseValidator.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.4.19;
+
+import './TokenValidator.sol';
+import '../../node_modules/zeppelin-solidity/contracts/math/SafeMath.sol';
+
+/*
+  Synchronous close validtor
+  All tokens have the same hold period and start trading on the same day
+
+         startTime    endTime    endTime + holdPeriod    dealOver
+           -tX         t0                tY                 tZ
+            |==========|=================|==================|
+Prehistory  |   Mint   |       Hold      |       Trade      |
+
+ */
+contract PhaseValidator is TokenValidator {
```

Comment:
NOTE TO SELF:

This actually makes sense in its own contract. We are probably going to have situations where certain users are exempt from certain phases and whatnot. Keeping these decomposed gives us the flexibility to call the parts that we need to. `modifier`s hardcode that logic per contract class, making it more difficult to extend. Broadly, this is good OO practice, but feels weird knowing this much about the contract in the validator.